flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/LabeledPointWithWeightGenerator.java

```diff
@@ -0,0 +1,194 @@
+        return get(WEIGHT_ARITY);
+    }
+
+    public LabeledPointWithWeightGenerator setWeightArity(int value) {
+        return set(WEIGHT_ARITY, value);
+    }
+
+    @Override
+    public Table[] getData(StreamTableEnvironment tEnv) {
+        StreamExecutionEnvironment env = TableUtils.getExecutionEnvironment(tEnv);
+
+        DataStream<Row> dataStream =
+                env.fromParallelCollection(
+                                new NumberSequenceIterator(1L, getNumValues()),
+                                BasicTypeInfo.LONG_TYPE_INFO)
+                        .map(
+                                new RandomLabeledPointWithWeightGenerator(
+                                        getSeed(),
+                                        getVectorDim(),
+                                        getFeatureArity(),
+                                        getLabelArity(),
+                                        getWeightArity()),
+                                new RowTypeInfo(
+                                        new TypeInformation[] {
+                                            DenseVectorTypeInfo.INSTANCE, Types.DOUBLE, Types.DOUBLE
+                                        },
+                                        new String[] {
+                                            getFeaturesCol(), getLabelCol(), getWeightCol()
+                                        }));
+
+        Table dataTable = tEnv.fromDataStream(dataStream);
+
+        return new Table[] {dataTable};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    private static class RandomLabeledPointWithWeightGenerator extends RichMapFunction<Long, Row> {
+        private final long initSeed;
+        private final int vectorDim;
+        private final int featureArity;
+        private final int labelArity;
+        private final int weightArity;
+        private Random random;
+
+        private RandomLabeledPointWithWeightGenerator(
+                long initSeed, int vectorDim, int featureArity, int labelArity, int weightArity) {
+            this.initSeed = initSeed;
+            this.vectorDim = vectorDim;
+            this.featureArity = featureArity;
+            this.labelArity = labelArity;
+            this.weightArity = weightArity;
+        }
+
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            super.open(parameters);
+            int index = getRuntimeContext().getIndexOfThisSubtask();
+            random = new Random(Tuple2.of(initSeed, index).hashCode());
+        }
+
+        @Override
+        public Row map(Long ignored) {
+            double[] features = new double[vectorDim];
+            for (int i = 0; i < vectorDim; i++) {
+                features[i] = getValue(featureArity, 0);
+            }
+
+            double label = getValue(labelArity, 0);
+
+            double weight = getValue(weightArity, 1);
```

Comment:
By setting `discreteValueOffset = 1`, when weightArity = 0, the generated weight value would be in the range [1, 2). This appears to be inconsistent with the weightArity specification.

Reply:
This comment is not applicable now.